internal/cortex/frontend/transport/handler.go

```diff
@@ -122,14 +147,54 @@ func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	r.Body = http.MaxBytesReader(w, r.Body, f.cfg.MaxBodySize)
 	r.Body = io.NopCloser(io.TeeReader(r.Body, &buf))
 
+	if f.cfg.EnableFailedQueryCache {
+		// Store query expression
+		queryExpressionNormalized = f.regex.ReplaceAllString(r.URL.Query().Get("query"), " ")
+
+		// Time range length for queries, if either of "start" or "end" are not present, return 0
+		getQueryRangeSeconds := func() int {
```

Comment:
please consider the step param from query_range API, it also matters to the expensiveness, increasing steps will reduce data fetched from storage backend.

Reply:
I agree `step` has an impact to query expansiveness, but it's not obvious for how to factor it in to calculate expansiveness. Since it's an intern starter project, I'd prefer keeping it simple and leaving `step` out for now. We can iterate on the expansiveness calculation later.

Reply:
sgtm